clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -947,12 +947,11 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
           // Other kinds of vectors.  Element-wise comparison returning
           // a vector.
           cir::CmpOpKind Kind = ClangCmpToCIRCmp(E->getOpcode());
-          return Builder.create<cir::VecCmpOp>(CGF.getLoc(BOInfo.Loc),
-                                               CGF.getCIRType(BOInfo.FullType),
-                                               Kind, BOInfo.LHS, BOInfo.RHS);
+          Result = Builder.create<cir::VecCmpOp>(
+              CGF.getLoc(BOInfo.Loc), CGF.getCIRType(BOInfo.FullType), Kind,
+              BOInfo.LHS, BOInfo.RHS);
         }
```

Comment:
It seems there's no change in vector related testcases. Should we also fix the result type of `vec.cmp` (if any problem)?